TRENZ.Lib.RazorMail.Core/Models/MailHeaderCollection.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace TRENZ.Lib.RazorMail.Models;
+
+/// <summary>
+/// Represents a collection of mail headers.
+/// </summary>
+public class MailHeaderCollection : Dictionary<string, object>
+{
+    /// <summary>
+    /// The recipients of the mail message.
+    /// </summary>
+    public IEnumerable<MailAddress> Recipients
+    {
+        get => GetAddresses("To");
+        set => SetAddresses("To", value);
+    }
+
+    /// <summary>
+    /// Recipients who will receive a copy of the mail message.
+    /// </summary>
+    public IEnumerable<MailAddress> CarbonCopy
+    {
+        get => GetAddresses("CC");
+        set => SetAddresses("CC", value);
+    }
+
+    /// <summary>
+    /// Recipients who will receive a blind carbon copy of the mail message.
+    /// </summary>
+    public IEnumerable<MailAddress> BlindCarbonCopy
+    {
+        get => GetAddresses("BCC");
+        set => SetAddresses("BCC", value);
+    }
+
+    /// <summary>
+    /// Addresses that will be used for replies to the mail message.
+    /// </summary>
+    public IEnumerable<MailAddress> ReplyTo
+    {
+        get => GetAddresses("Reply-To");
+        set => SetAddresses("Reply-To", value);
+    }
+
+    /// <summary>
+    /// The sender of the mail message.
+    /// </summary>
+    public MailAddress? From
+    {
+        get
+        {
+            if (ContainsKey("From"))
```

Comment:
shouldn't these use `TryGetValue` to avoid double lookup?

Reply:
Yeah, fixed